dask/array/core.py

```diff
@@ -378,6 +378,10 @@ def rec_concatenate(arrays, axis=0):
         arrays = [a[None, ...] for a in arrays]
     if len(arrays) == 1:
         return arrays[0]
+    for a in arrays:
+        if hasattr(a, 'shape'):
+            if all(i == 0 for i in a.shape):
+                arrays.remove(a)
```

Comment:
Okay I had a separate issue in my environment that was causing problems sorry. There is only one broken test. Slicing can creates arrays with zero size, and they use `rec_concatenate` so `test_slicing.py::test_empty_slice` breaks.